File: jsimple-json/src/main/java/jsimple/json/readerwriter/JsonIntProperty.java
package jsimple.json.readerwriter;

/**
 * @author Bret Johnson
 * @since 5/17/13 12:40 AM
 */
public class JsonIntProperty extends JsonProperty {
    public JsonIntProperty(String name, int id) {
        super(name, id);
    }

    public int readValue(JsonObjectReader objectReader) {
        return (Integer) objectReader.readPrimitiveValue();
    }

    public void write(JsonObjectWriter objectWriter, int value) {
        objectWriter.write(this, value);
    }
}

File: jsimple-json/c#/jsimple/json/readerwriter/JsonIntProperty.cs
namespace jsimple.json.readerwriter
{

	/// <summary>
	/// @author Bret Johnson
	/// @since 5/17/13 12:40 AM
	/// </summary>
	public class JsonIntProperty : JsonProperty
	{
		public JsonIntProperty(string name, int id) : base(name, id)
		{
		}

		public virtual int readValue(JsonObjectReader objectReader)
		{
			return (int?) objectReader.readPropertyValue();
		}

		public override object readValueUntyped(JsonObjectReader objectReader)
		{
			return objectReader.readPropertyValue();
		}

		public virtual void write(JsonObjectWriter objectWriter, int value)
		{
			objectWriter.write(this, value);
		}
	}

}